bestbook/static/build/js/index.js

```diff
@@ -188,12 +190,109 @@ $( function() {
       for(var i = 0; i < candidates.length; ++i) {
         formData.candidates.push(candidates[i].olid)
       }
-      // TODO: Handle success and failure
-      $.ajax({
-        type: 'POST',
-        url: '/submit',
-        data: formData,
-      })
+
+      let $description = $('#description');
+      // Sets whitespace only description to an empty string, 
+      // which will trigger UI hint from browser
+      $description.val($description.val().trim());
+      formData.description = $description.val();
+
+      if(validateRecommendationFormData()) {
+        // TODO: Handle failure cases (server error)
+        $.ajax({
+          type: 'POST',
+          url: '/submit',
+          data: formData,
+          success: function() {
+            window.location = recommendation_redirect_url;
+          }
+        })
```

Comment:
could be moved to a separate function outside main loop (especially if we may plan on invoking this from different contexts), but not a blocker.

Easy to refactor at a later point